client/coral-framework/services/bootstrap.js

```diff
@@ -178,6 +178,10 @@ export async function createContext({
     await preInit(context);
   }
 
+  if (init) {
+    await init(context);
+  }
+
   // Run initialization.
   await Promise.all([init, plugins.executeInit(context)]);
```

Comment:
It does work like you did. Like this it first runs and wait for `init` and then runs and wait for plugins init. It'd be more efficient to run both `init` and `plugins.executeInit` in one step and then wait for their promises to resolve.

E.g.

```js
import noop from 'lodash/noop';

export async function createContext({
   [...]
   init = noop,
 } = {}) {
  [...]
  await Promise.all([init(context), plugins.executeInit(context)]);
}
```